### PR TITLE
Tensorizer checks mounted directories and doesn't save empty df

### LINKED ISSUE #976: Update tensorizer to not create untensorized_bedmaster_files.csv if all files are tensorized
## Why it Matters
Even if no untensorized Bedmaster files are logged by the end of tensorization, the pipeline still creates an empty `untensorized_bedmaster_files.csv` file.

## Possible Solution
If empty, don't create it.

### LINKED ISSUE #1194: Tensorization checks if desired file locations are mounted
## What and why
To avoid a script running and not throwing an error when the user forgot to mount the required network share.

## Solution(s)
Check if desired file locations are mounted. If not, throw error.

## Acceptance criteria
If `tensorize` requires data from a specific mounted network share that is not mounted during runtime, a descriptive error is thrown.

### LINKED ISSUE #1213: Explore crashes if there is no intersection
## Error message or warning

```
2021-01-25 13:50:58 - ERROR - recipes:129 - '<=' not supported between instances of 'str' and 'float'
Traceback (most recent call last):
  File "/home/rp416/repos/ml4c3/ml4c3/recipes.py", line 87, in run
    explore(args=args, disable_saving_output=args.explore_disable_saving_output)
  File "/usr/local/lib/python3.8/dist-packages/ml4c3/explorations.py", line 518, in explore
    stats = _calculate_summary_stats(
  File "/usr/local/lib/python3.8/dist-packages/ml4c3/explorations.py", line 698, in _calculate_summary_stats
    stats["min"] = df[key].min()
  File "/usr/local/lib/python3.8/dist-packages/pandas/core/generic.py", line 11204, in min
    return NDFrame.min(self, axis, skipna, level, numeric_only, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pandas/core/generic.py", line 10709, in min
    return self._stat_function(
  File "/usr/local/lib/python3.8/dist-packages/pandas/core/generic.py", line 10704, in _stat_function
    return self._reduce(
  File "/usr/local/lib/python3.8/dist-packages/pandas/core/series.py", line 4152, in _reduce
    return op(delegate, skipna=skipna, **kwds)
  File "/usr/local/lib/python3.8/dist-packages/pandas/core/nanops.py", line 133, in f
    result = alt(values, axis=axis, skipna=skipna, **kwds)
  File "/usr/local/lib/python3.8/dist-packages/pandas/core/nanops.py", line 392, in new_func
    result = func(values, axis=axis, skipna=skipna, mask=mask, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pandas/core/nanops.py", line 975, in reduction
    result = getattr(values, meth)(axis)
  File "/usr/local/lib/python3.8/dist-packages/numpy/core/_methods.py", line 34, in _amin
    return umr_minimum(a, axis, None, out, keepdims, initial, where)
TypeError: '<=' not supported between instances of 'str' and 'float'
2021-01-25 13:50:58 - INFO - recipes:136 - Executed explore operation in 1597.03 sec
```

## Expected behavior
This happens when there is no intersection of MRNs having all the tensors.

## Possible solution
If there is no intersection it does not crash.
Also, it will be great if it returns a .csv file where each row corresponds to a MRN/input file and each column if it returned a valid tensor or not (True or False) for the corresponding tensor map.

## Steps to reproduce
1. Run `~/dropbox/cardiac-arrest/scripts/explore.sh`

### PR DESCRIPTION
**Pull request by raimonpv**
## Issues closed
Closes #1194 
Closes #976 

## Changelog
- [x]  If unfinished bedmaster files is empty is not saved.
- [x] Checks edw and bedmaster folders before tensorization

## Checklist
- [x ] The following are complete:
    - Added or revised tests if appropriate.
    - If new packages are installed, `requirements.txt` is updated.
    - If documentation should be updated, a new issue was made in `ml4c3-wiki`.
    - The PR name describes its content and does not contain "Ref #".
    - The PR does not add PHI.

### CODE REVIEW

**Comment by erikr** [comment]
@raimonpv you have had a really productive day! Wow, thanks for handling all of these infra issues!

Would you consider breaking this up into three smaller PRs? The title here suggests it is perhaps overly broad. The reason is that it can be helpful in the future to have more granular, precise PRs when we try to figure out the exact code behind a functional change.

File: ingest/icu/tensorizer.py [hunk 1]
```diff
@@ -348,6 +349,14 @@ def _copy_hd5(staging_dir, destination_dir, file):
 
 
 def tensorize(args):
+
+    if not os.path.isdir(args.bedmaster) or len(os.listdir(args.bedmaster)) == 0:
+        raise ValueError(
+            f"CDAC server with Bedmaster data is not mounted in {args.bedmaster}",
```
Comment (erikr) [inline ingest/icu/tensorizer.py:355]: Generalize beyond CDAC:
```suggestion
            f"Server with Bedmaster data is not mounted in {args.bedmaster}",
```

File: ingest/icu/tensorizer.py [hunk 2]
```diff
@@ -348,6 +349,14 @@ def _copy_hd5(staging_dir, destination_dir, file):
...
+        )
+    if not os.path.isdir(args.edw) or len(os.listdir(args.edw)) == 0:
+        raise ValueError(f"MAD3 server with EDW data is not mounted in {args.edw}")
```
Comment (erikr) [inline ingest/icu/tensorizer.py:358]: Generalize:
```suggestion
        raise ValueError(f"Server with EDW data is not mounted in {args.edw}")
```

File: tensorize/bedmaster/tensorizer.py [hunk 3]
```diff
@@ -274,6 +275,12 @@ def _copy_hd5(staging_dir, destination_dir, file):
 
 
 def tensorize(args):
+
+    if not os.path.isdir(args.bedmaster) or len(os.listdir(args.bedmaster)) == 0:
+        raise ValueError(
+            f"CDAC server with Bedmaster data is not mounted in {args.bedmaster}",
```
Comment (erikr) [inline tensorize/bedmaster/tensorizer.py:281]: same comment as above

File: tensorize/edw/tensorizer.py [hunk 4]
```diff
@@ -255,6 +255,10 @@ def _copy_hd5(staging_dir, destination_dir, file):
 
 
 def tensorize(args):
+
+    if not os.path.isdir(args.edw) or len(os.listdir(args.edw)) == 0:
+        raise ValueError(f"MAD3 server with EDW data is not mounted in {args.edw}")
```
Comment (erikr) [inline tensorize/edw/tensorizer.py:260]: same comment as above

**Review by erikr — APPROVED** [review]
Thanks!

Minor comments about naming.

I see redundancy between `/tensorize/edw/$SCRIPT.py` and `/tensorize/icu/$SCRIPT.py` that we should resolve in a future PR.

File: tensorize/bedmaster/tensorizer.py (hunk 3 above)
Comment (raimonpv) [inline tensorize/bedmaster/tensorizer.py:281]:
```suggestion
            f"Server with Bedmaster data is not mounted in {args.bedmaster}",
```

File: tensorize/edw/tensorizer.py (hunk 4 above)
Comment (raimonpv) [inline tensorize/edw/tensorizer.py:260]:
```suggestion
        raise ValueError(f"Server with EDW data is not mounted in {args.edw}")
```

**Comment by raimonpv** [comment]
> I see redundancy between /tensorize/edw/$SCRIPT.py and /tensorize/icu/$SCRIPT.py that we should resolve in a future PR.

Should be tackled in https://github.com/aguirre-lab/ml4c3/issues/993

**Pull request merged**